packages/material-ui/src/Select/Select.d.ts

```diff
@@ -142,7 +142,7 @@ export interface SelectProps
    * @param {any} value The `value` provided to the component.
    * @returns {ReactNode}
    */
-  renderValue?: (value: SelectProps['value']) => React.ReactNode;
+  renderValue?: (value: SelectProps<T>['value']) => React.ReactNode;
```

Comment:
Would this work now?
```suggestion
  renderValue?: (value: T) => React.ReactNode;
```